channel/erc20_depositor.go

```diff
@@ -40,42 +45,138 @@ const ERC20DepositorTXGasLimit = 100000
 // Return value of ERC20Depositor.NumTx.
 const erc20DepositorNumTx = 2
 
+// Keep track of the increase allowance and deposit processes.
+var mu sync.Mutex
+var locks = make(map[string]*sync.Mutex)
+
+// DepositResult is created to keep track of the returned values.
+type DepositResult struct {
+	Transactions types.Transactions
+	Error        error
+}
+
+// Create key from account address and asset to only lock the process when hub deposits the same asset at the same time.
+func lockKey(account common.Address, asset common.Address) string {
```

Comment:
I would prefer to move all these helper functions after the `ERC20Depositor` methods of this file, so the first method would be the constructor.